flang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -832,12 +833,15 @@ static bool parseSemaArgs(CompilerInvocation &res, llvm::opt::ArgList &args,
   // -J/module-dir option
   auto moduleDirList =
       args.getAllArgValues(clang::driver::options::OPT_module_dir);
-  // User can only specify -J/-module-dir once
+  // User can only specify one -J/-module-dir directory
   // https://gcc.gnu.org/onlinedocs/gfortran/Directory-Options.html
+  std::sort(moduleDirList.begin(), moduleDirList.end());
+  moduleDirList.erase(std::unique(moduleDirList.begin(), moduleDirList.end()),
+                      moduleDirList.end());
   if (moduleDirList.size() > 1) {
     const unsigned diagID =
         diags.getCustomDiagID(clang::DiagnosticsEngine::Error,
-                              "Only one '-module-dir/-J' option allowed");
+                              "Only one '-module-dir/-J' directory allowed");
```

Comment:
This is not 100% clear that the number of -J isn't the problem, it's the number of unique directories given. How about:
"Only one `-module-dir/-J' directory allowed. '-module-dir/-J' may be given multiple times but the directory must be the same each time."

It's wordy but less room for confusion.

Reply:
Ofc someone will wonder why this is allowed at all, but this is not worth explaining in the error message.

Reply:
Changed according to your suggestion, thanks.